presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -1065,6 +1064,24 @@ private Type coerceToSingleType(StackableAstVisitorContext<AnalysisContext> cont
         }
     }
 
+    public static Signature resolveFunction(FunctionCall node, boolean isApproximate, List<TypeSignature> argumentTypes, FunctionRegistry functionRegistry)
```

Comment:
I extracted the function resolution code into a method so that I can re-use it when rewriting the `GroupingOperation` to a `FunctionCall` and ensure that the `grouping()` function exists and if it doesn't fail before execution.